src/Codec/Xlsx/Types/DataValidation.hs

```diff
@@ -0,0 +1,200 @@
+    | ValidationTypeDecimal    OperatorExpression
+    | ValidationTypeList       [Text]
+    | ValidationTypeTextLength OperatorExpression
+    | ValidationTypeTime       OperatorExpression
+    | ValidationTypeWhole      OperatorExpression
+    deriving (Eq, Show)
+
+-- See 18.18.18 "ST_DataValidationErrorStyle (Data Validation Error Styles)" (p. 2438/2448)
+data ErrorStyle
+    = ErrorStyleInformation
+    | ErrorStyleStop
+    | ErrorStyleWarning
+    deriving (Eq, Show)
+
+-- See 18.3.1.32 "dataValidation (Data Validation)" (p. 1614/1624)
+data DataValidation = DataValidation
+    { _dvAllowBlank       :: Maybe Bool
+    , _dvError            :: Maybe Text
+    , _dvErrorStyle       :: Maybe ErrorStyle
+    , _dvErrorTitle       :: Maybe Text
+    , _dvPrompt           :: Maybe Text
+    , _dvPromptTitle      :: Maybe Text
+    , _dvShowDropDown     :: Maybe Bool
+    , _dvShowErrorMessage :: Maybe Bool
+    , _dvShowInputMessage :: Maybe Bool
+    , _dvSqref            :: SqRef
+    , _dvValidationType   :: Maybe ValidationType
+    } deriving (Eq, Show)
+
+makeLenses ''DataValidation
+
+instance Default DataValidation where
+    def = DataValidation
+      Nothing Nothing Nothing Nothing Nothing Nothing Nothing Nothing Nothing
+      (SqRef []) Nothing
+
+{-------------------------------------------------------------------------------
+  Parsing
+-------------------------------------------------------------------------------}
+
+instance FromAttrVal ErrorStyle where
+    fromAttrVal "information" = readSuccess ErrorStyleInformation
+    fromAttrVal "stop"        = readSuccess ErrorStyleStop
+    fromAttrVal "warning"     = readSuccess ErrorStyleWarning
+    fromAttrVal t             = invalidText "ErrorStyle" t
+
+instance FromCursor DataValidation where
+    fromCursor cur = cur $/ element (n_ "dataValidation") >=> fromCur
+      where
+        fromCur c = do
+            _dvAllowBlank       <- maybeAttribute "allowBlank"       c
+            _dvError            <- maybeAttribute "error"            c
+            _dvErrorStyle       <- maybeAttribute "errorStyle"       c
+            _dvErrorTitle       <- maybeAttribute "errorTitle"       c
+            mop                 <- maybeAttribute "operator"         c
+            _dvPrompt           <- maybeAttribute "prompt"           c
+            _dvPromptTitle      <- maybeAttribute "promptTitle"      c
+            _dvShowDropDown     <- maybeAttribute "showDropDown"     c
+            _dvShowErrorMessage <- maybeAttribute "showErrorMessage" c
+            _dvShowInputMessage <- maybeAttribute "showInputMessage" c
+            _dvSqref            <- fromAttribute  "sqref"            c
+            mtype               <- maybeAttribute "type"             c
+            _dvValidationType   <- readValidation mop mtype          c
+            return DataValidation{..}
+
+readValidation :: Maybe Text -> Maybe Text -> Cursor -> [Maybe ValidationType]
```

Comment:
this name is somewhat confusing, let's use something more explicit, e.g. `readValidationType` maybe?